core/drivers/stm32_tamp.c

```diff
@@ -133,31 +161,184 @@ struct stm32_tamp_compat {
+				_TAMP_CIDCFGR_CONF_MASK,
+				stm32_tamp_dev->pdata.conf_data->cid_confs[i]);
+	}
+}
 
-TEE_Result stm32_tamp_set_secure_bkpregs(struct stm32_bkpregs_conf *bkr_conf)
+static TEE_Result stm32_tamp_apply_bkpr_rif_conf(void)
 {
-	struct stm32_tamp_instance *tamp = stm32_tamp_device;
+	struct stm32_bkpregs_conf *bkpregs_conf =
+			stm32_tamp_dev->pdata.bkpregs_conf;
+	vaddr_t base = io_pa_or_va(&stm32_tamp_dev->pdata.base, 1);
+	int i = 0;
+
+	if (!bkpregs_conf)
+		panic("No backup register configuration");
+
+	for (i = 0; i < 4; i++) {
+		if (bkpregs_conf->rif_offsets[i] >
+		    (stm32_tamp_dev->hwconf1 & _TAMP_HWCFGR1_BKPREG))
+			return TEE_ERROR_NOT_SUPPORTED;
+	}
+
+	/* Fill the 3 TAMP_BKPRIFRx registers */
+	io_clrsetbits32(base + _TAMP_BKPRIFR(1), _TAMP_BKPRIFR_1_MASK,
+			bkpregs_conf->rif_offsets[0]);
+	io_clrsetbits32(base + _TAMP_BKPRIFR(2), _TAMP_BKPRIFR_2_MASK,
+			bkpregs_conf->rif_offsets[1]);
+	io_clrsetbits32(base + _TAMP_BKPRIFR(3), _TAMP_BKPRIFR_3_MASK,
+			bkpregs_conf->rif_offsets[2] |
+			(bkpregs_conf->rif_offsets[3] <<
+			 _TAMP_BKPRIFR_ZONE3_RIF2_SHIFT));
+
+	DMSG("Backup registers mapping :");
+	DMSG("********START of zone 1********");
+	DMSG("Protection Zone 1-RIF1 begins at register: 0");
+	DMSG("Protection Zone 1-RIF2 begins at register: %"PRIu32,
+	     bkpregs_conf->rif_offsets[0]);
+	DMSG("Protection Zone 1-RIF2 ends at register: %"PRIu32,
+	     bkpregs_conf->nb_zone1_regs ? bkpregs_conf->nb_zone1_regs - 1 : 0);
+	DMSG("********END of zone 1********");
+	DMSG("********START of zone 2********");
+	DMSG("Protection Zone 2-RIF1 begins at register: %"PRIu32,
+	     bkpregs_conf->nb_zone1_regs);
+	DMSG("Protection Zone 2-RIF2 begins at register: %"PRIu32,
+	     bkpregs_conf->rif_offsets[1]);
+	DMSG("Protection Zone 2-RIF2 ends at register: %"PRIu32,
+	     bkpregs_conf->rif_offsets[1] > bkpregs_conf->nb_zone1_regs ?
+	     bkpregs_conf->nb_zone2_regs - 1 : 0);
+	DMSG("********END of zone 2********");
+	DMSG("********START of zone 3********");
+	DMSG("Protection Zone 3-RIF1 begins at register: %"PRIu32,
+	     bkpregs_conf->nb_zone2_regs);
+	DMSG("Protection Zone 3-RIF0 begins at register: %"PRIu32,
+	     bkpregs_conf->rif_offsets[2]);
+	DMSG("Protection Zone 3-RIF2 begins at register: %"PRIu32,
+	     bkpregs_conf->rif_offsets[3]);
+	DMSG("Protection Zone 3-RIF2 ends at the last register: 128");
+	DMSG("********END of zone 3********");
+
+	return TEE_SUCCESS;
+}
+
+/*
+ * stm32_tamp_set_secure_bkprwregs: Configure backup registers zone.
+ * registers in zone 1: read/write only in secure mode
+ *              zone 2: write only in secure mode, read in secure and
+ *                       non-secure mode
+ *              zone 3: read/write in secure and non-secure mode
+ *
+ * return TEE_ERROR_NOT_SUPPORTED:  if zone 1 and/or zone 2 definition are out
+ *                                  of range.
+ *        TEE_ERROR_BAD_PARAMETERS: if bkpregs_cond is NULL.
+ *        TEE_SUCCESS             : if OK.
+ */
```

Comment:
Siince it's now a local function, maybe you don't need this detailed description.